src/entry.c

```diff
@@ -404,7 +471,5 @@ entry *entryDefrag(entry *entry, void *(*defragfn)(void *), sds (*sdsdefragfn)(s
  * forked and memory won't be used again. See zmadvise_dontneed() */
 void entryDismissMemory(entry *entry) {
     /* Only dismiss values memory since the field size usually is small. */
-    if (entryHasValuePtr(entry)) {
-        dismissSds(*entryGetValueRef(entry));
-    }
+    if (entryHasValuePtr(entry)) entryFreeValuePtr(entry);
```

Comment:
This function needs review.  It's not clear that it's achieving the stated objectives.

In the previous code, `dismissSds` is called.  This is a function inside of `object.c` - and it doesn't seem that we should be taking a dependency to `object.c` in this code.  The function itself is just a trivial wrapper on `sdsfree` - which would seem to be the more appropriate function.

The comment talks about this reducing CoW during a fork.  However, the act of releasing a small string in a memory bin is unlikely to have any affect on madvise.  And it might actually dirty the page by the act of freeing - initiating a CoW operation.

Recommend reviewing the logic/need for this function.  When is this used rather than `entryFree`?